packages/dev/core/src/Meshes/csg2.ts

```diff
@@ -0,0 +1,513 @@
+        for (let i = 0; i < sourceIndices.length; i += 3) {
+            triVerts[i] = sourceIndices[i + 2];
+            triVerts[i + 1] = sourceIndices[i + 1];
+            triVerts[i + 2] = sourceIndices[i];
+        }
+
+        // Positions
+        const tempVector3 = new Vector3();
+        let numProp = 3;
+        const structure: IManifoldVertexComponent[] = [{ stride: 3, kind: VertexBuffer.PositionKind }];
+
+        if (ignoreWorldMatrix) {
+            structure[0].data = sourceVertices;
+        } else {
+            const positions = new Float32Array(sourceVertices.length);
+            for (let i = 0; i < sourceVertices.length; i += 3) {
+                Vector3.TransformCoordinatesFromFloatsToRef(sourceVertices[i], sourceVertices[i + 1], sourceVertices[i + 2], worldMatrix, tempVector3);
+                tempVector3.toArray(positions, i);
+            }
+            structure[0].data = positions;
+        }
+
+        // Normals
+        const sourceNormals = mesh.getVerticesData(VertexBuffer.NormalKind);
+        if (sourceNormals) {
+            numProp += 3;
+            structure.push({ stride: 3, kind: VertexBuffer.NormalKind });
+            if (ignoreWorldMatrix) {
+                structure[1].data = sourceNormals;
+            } else {
+                const normals = new Float32Array(sourceNormals.length);
+                for (let i = 0; i < sourceNormals.length; i += 3) {
+                    Vector3.TransformNormalFromFloatsToRef(sourceNormals[i], sourceNormals[i + 1], sourceNormals[i + 2], worldMatrix, tempVector3);
+                    tempVector3.toArray(normals, i);
+                }
+                structure[1].data = normals;
+            }
+        }
+
+        // UVs
+        for (const kind of [VertexBuffer.UVKind, VertexBuffer.UV2Kind, VertexBuffer.UV3Kind, VertexBuffer.UV4Kind, VertexBuffer.UV5Kind, VertexBuffer.UV6Kind]) {
+            const sourceUV = mesh.getVerticesData(kind);
+            if (sourceUV) {
+                numProp += 2;
+                structure.push({ stride: 2, kind: kind, data: sourceUV });
+            }
+        }
+
+        // Colors
+        const sourceColors = mesh.getVerticesData(VertexBuffer.ColorKind);
+        if (sourceColors) {
+            numProp += 4;
+            structure.push({ stride: 4, kind: VertexBuffer.ColorKind, data: sourceColors });
+        }
+
+        // Process
+        return this._ProcessData(sourceVertices.length / 3, triVerts, structure, numProp, runIndex, runOriginalID);
+    }
+}
+
+/**
+ * Checks if the Manifold library is ready
+ * @returns true if the Manifold library is ready
+ */
+export function IsCSG2Ready() {
+    return Manifold !== undefined;
+}
+
+/**
+ * Initialize the Manifold library
+ * @param options defines the options to use to initialize the library
+ */
+export async function InitializeCSG2Async(options?: Partial<ICSG2Options>) {
+    const localOptions = {
+        manifoldUrl: "https://unpkg.com/manifold-3d@2.5.1",

```

Comment:
do we want to deploy this to our CDN? we had bad experience with unpkg (or was it jsdeliver?) in the past. Eventually we have decided to only reference files from our CDN

Reply:
I do not want us to have this responsibility anymore. This could be a good way to encourage users to NOT use our CDN for prod ;)